src/gobupload/storage/update_table.py

```diff
@@ -0,0 +1,612 @@
+                f'dst.{FIELD.SEQNR} AS dst_volgnummer'
+            )
+
+        distinct_seqnr = f'OR dst.{FIELD.SEQNR} IS DISTINCT FROM rel.dst_volgnummer'
+
+        select_expressions.append(f"""CASE
+        WHEN rel.src_id IS NULL THEN '{ADD.name}'
+        WHEN src.{FIELD.ID} IS NULL THEN '{DELETE.name}'
+        WHEN dst.{FIELD.ID} IS DISTINCT FROM rel.dst_id
+            {distinct_seqnr if self.dst_has_states else ''}
+             OR LEAST(src.{FIELD.EXPIRATION_DATE}, dst.{FIELD.EXPIRATION_DATE})
+             IS DISTINCT FROM rel.{FIELD.EXPIRATION_DATE}
+             THEN '{MODIFY.name}'
+        ELSE '{CONFIRM.name}'
+    END AS event_type""")
+        return select_expressions
+
+    def _source_value_ref(self):
+        """Returns the reference to the source value in the src object. For a many relation this reference points to
+        the unpacked JSONB object.
+
+        :return:
+        """
+
+        return f"{self.json_join_alias}.item->>'{FIELD.SOURCE_VALUE}'" if self.is_many \
+            else f"src.{self.src_field_name}->>'{FIELD.SOURCE_VALUE}'"
+
+    def _rel_table_join_on(self):
+        """Returns the ON clause for the relation table join
+
+        :return:
+        """
+        join_on = [f"src.{FIELD.ID} = rel.src{FIELD.ID}"]
+        if self.src_has_states:
+            join_on.append(f"src.{FIELD.SEQNR} = rel.src_{FIELD.SEQNR}")
+
+        join_on.append(f'rel.src_source = src.{FIELD.SOURCE}')
+        join_on.append(f'rel.{FIELD.DATE_DELETED} IS NULL')
+        join_on.append(f'rel.bronwaarde = {self._source_value_ref()}')
+        join_on.append(f'rel.{FIELD.APPLICATION} = src.{FIELD.APPLICATION}')
+        return join_on
+
+    def _geo_resolve(self, spec, src_ref='src'):
+        src_geo = f"{src_ref}.{spec['source_attribute']}"
+        dst_geo = f"dst.{spec['destination_attribute']}"
+
+        resolvers = {
+            LIES_IN: f"ST_Contains({dst_geo}::geometry, ST_PointOnSurface({src_geo}::geometry))"
+        }
+        return resolvers.get(spec["method"])
+
+    def _json_obj_ref(self, src_ref='src'):
+        return f'{self.json_join_alias}.item' if self.is_many else f'{src_ref}.{self.src_field_name}'
+
+    def _relate_match(self, spec, src_ref='src'):
+        if spec['method'] == EQUALS:
+            return f"dst.{spec['destination_attribute']} = {self._json_obj_ref(src_ref)}->>'bronwaarde'"
+        else:
+            return self._geo_resolve(spec, src_ref)
+
+    def _dst_table_inner_join_on(self, src_ref='src'):
+        """Returns the ON clause for the dst table join
+
+        :param src_ref:
+        :return:
+        """
+        join_on = ([f"({src_ref}.{FIELD.APPLICATION} = '{spec['source']}' AND " +
+                    f"{self._relate_match(spec, src_ref)})" for spec in self.relation_specs])
+        # If more matches have been defined that catch any of the matches
+        if len(join_on) > 1:
+            join_on = [f"({self.or_join.join(join_on)})"]
```

Comment:
I don't get this statement. You set join_on to ([]) and then test len?

Reply:
Never copy code from Vegter haha

Reply:
But still the code remains a little strange. You assign (or I assigned) join_on and then test len(join_on). Well this is an easy test, it is always true

Reply:
Can't join_on be of length 1?